pkg/machine/wsl/wutil/wutil.go

```diff
@@ -5,14 +5,69 @@ package wutil
 import (
 	"bufio"
 	"io"
+	"os"
 	"os/exec"
+	"path/filepath"
 	"strings"
+	"sync"
 	"syscall"
 
 	"golang.org/x/text/encoding/unicode"
 	"golang.org/x/text/transform"
 )
 
+var (
+	once    sync.Once
+	wslPath string
+)
+
+func FindWSL() string {
+	// At the time of this writing, a defect appeared in the OS preinstalled WSL executable
+	// where it no longer reliably locates the preferred Windows App Store variant.
+	//
+	// Manually discover (and cache) the wsl.exe location to bypass the problem
+	once.Do(func() {
+		var locs []string
+
+		// Prefer Windows App Store version
+		if localapp := getLocalAppData(); localapp != "" {
+			locs = append(locs, filepath.Join(localapp, "Microsoft", "WindowsApps", "wsl.exe"))
+		}
+
+		// Otherwise, the common location for the legacy system version
+		root := os.Getenv("SystemRoot")
+		if root == "" {
+			root = `C:\Windows`
```

Comment:
would there be any reason to use something like a known command and then examine from that?